pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/inv/text/LuceneFSTIndexCreator.java

```diff
@@ -21,11 +21,14 @@
 import java.io.File;
```

Comment:
Is there a reason this class does not extend AbstractTextIndexCreator.java?

Reply:
We already chat about that, but I want to drop these lines here in order to inform the others:

FST index and Text index are two completely different things that by index String columns. In some parts of the code we consider them two flavours of the same thing and in other we consider them different things. In my opinion, they are actually very different things, they actually change the semantics of the queries when one or the other index the column (match behaves different!).

I would like to split them even more in this PR, but I tried to apply the less possible changes (which is funny to say when the PR changes 10k lines!). In this particular case, the IndexCreators are completely different. The older implementation of `SegmentColumnarIndexCreator` stored one map for text index creators and one map for fst index creators. The method `SegmentColumnarIndexCreator.addRow` did called `add` for each text index creator but did not call `add` for fst index creators. That is why in this PR both FSTIndexCreators implement the add methods as a NOOP.

I insist that I strongly believe that we should totally split FST and text indexes, but I didn't tried to do that in this PR.

Reply:
Thanks for helping me understand this clearer. Might warrant a small comment somewhere in the two FST creator classes.

Reply:
Done. I tried to do the same with the reader, but seems more delicate, so I think we can do it in another PR